### PR TITLE
Atribukh/course endpoint

### PR DESCRIPTION
**Pull request by arstrybuh257**

### CODE REVIEW

File: UniAtHome/UniAtHome.WebAPI/Controllers/CourseController.cs [hunk 1]
```diff
@@ -32,38 +37,18 @@ public async Task<IActionResult> GetCourseById(int id)
...
-        public async Task<IActionResult> GetCourseByName(string name)
-        {
-            var courses = await this.courseService.GetCoursesByNameAsync(name);
-            if (courses != null)
-            {
-                return Ok(courses);
-            }
-
-            return BadRequest();
-        }
```
Comment (EdShel) [inline UniAtHome/UniAtHome.WebAPI/Controllers/CourseController.cs:67]: Why did you delete this method and didn't create any alternative one? It was fully operation at least until it was changed.

File: UniAtHome/UniAtHome.WebAPI/Controllers/UserController.cs [hunk 2]
```diff
@@ -106,19 +106,5 @@ public async Task<ObjectResult> Revoke()
...
-        {
-            return Ok();
-        }
-
-        [Authorize(Roles = "Admin")]
-        [HttpGet("checkAdmin")]
-        public ActionResult<string> CheckAdmin()
-        {
-            return Ok();
-        }
```
Comment (EdShel) [inline UniAtHome/UniAtHome.WebAPI/Controllers/UserController.cs]: And why did you delete these ones too? Testers must have an endpoint to test authorization & authentication.

File: UniAtHome/UniAtHome.BLL/Services/CourseService.cs [hunk 3]
```diff
@@ -7,38 +7,46 @@
...
 
         public async Task AddCourseAsync(CourseDTO courseDto)
         {
-            var course = this.mapper.Map<Course>(courseDto);
-            await this.repository.AddAsync(course);
+            var course = mapper.Map<Course>(courseDto);
+            var newCourse = await courseRepository.AddAsync(course);
+            await courseRepository.SaveChangesAsync();
+            var teacherId = (await userRepository.FindByEmailAsync(courseDto.TeacherEmail)).Id;
+            await courseMemberRepository.AddAsync(new CourseMember(teacherId, newCourse.Id));
```
Comment (lesloress) [inline UniAtHome/UniAtHome.BLL/Services/CourseService.cs]: I thought that we configured everything the way that there is no need to use courseMemberRepository and we can just use navigation property of our course entity (course.CourseMembers.Add(...))

File: UniAtHome/UniAtHome.BLL/Services/CourseService.cs [hunk 4]
```diff
@@ -7,38 +7,46 @@
...
+            await courseMemberRepository.SaveChangesAsync();
         }
 
         public async Task<IEnumerable<CourseDTO>> GetCoursesByNameAsync(string name)
         {
-            var courses = await repository.Find(
+            var courses = await courseRepository.Find(
                 course => EF.Functions.Like(course.Name, $"%{name}%"));
-            return courses.Select(course => this.mapper.Map<CourseDTO>(course));
+            return courses.Select(course => mapper.Map<CourseDTO>(course));
```
Comment (lesloress) [inline UniAtHome/UniAtHome.BLL/Services/CourseService.cs:46]: What is wrong with using this?

File: UniAtHome/UniAtHome.WebAPI/Controllers/CourseController.cs (hunk 1 above)
Comment (arstrybuh257) [reply UniAtHome/UniAtHome.WebAPI/Controllers/CourseController.cs:67]: I will create an alternative in the next PR. I will provide universal search endpoint.

File: UniAtHome/UniAtHome.WebAPI/Controllers/UserController.cs (hunk 2 above)
Comment (arstrybuh257) [reply UniAtHome/UniAtHome.WebAPI/Controllers/UserController.cs]: Argee. Fixed.

File: UniAtHome/UniAtHome.DAL/Repositories/Repository.cs [hunk 5]
```diff
@@ -34,27 +35,27 @@ public async Task SaveChangesAsync()
 
         public async Task<IEnumerable<T>> Find(Expression<Func<T, bool>> predicate)
         {
-            return await this.context.Set<T>().Where(predicate).AsNoTracking().ToListAsync();
+            return await context.Set<T>().Where(predicate).AsNoTracking().ToListAsync();
```
Comment (lesloress) [inline UniAtHome/UniAtHome.DAL/Repositories/Repository.cs:37]: we're back on that again.

File: UniAtHome/UniAtHome.BLL/Services/CourseService.cs (hunk 4 above)
Comment (arstrybuh257) [reply UniAtHome/UniAtHome.BLL/Services/CourseService.cs:46]: Nothing wrong, but resharper is angry about that.

File: UniAtHome/UniAtHome.BLL/Services/CourseService.cs (hunk 3 above)
Comment (arstrybuh257) [reply UniAtHome/UniAtHome.BLL/Services/CourseService.cs]: Fixed

**Review by lesloress — APPROVED** [review]
